brushfire-core/src/main/scala/com/stripe/brushfire/AnnotatedTree.scala

```diff
@@ -93,84 +122,84 @@ case class AnnotatedTree[K, V, T, A: Semigroup](root: Node[K, V, T, A]) {
   /**
    * Returns the leaf with index `leafIndex` by performing a DFS.
    */
-  def leafAt(leafIndex: Int): Option[LeafNode[K, V, T, A]] = leafAt(leafIndex, root)
+  def leafAt(leafIndex: Int): Option[LeafNode[K, V, T, A]] =
+    leafAt(leafIndex, root)
 
   /**
-   * Returns the leaf with index `leafIndex` that is a descendant of `start` by
-   * performing a DFS starting with `start`.
+   * Returns the leaf with index `leafIndex` that is a descendant of
+   * `start` by performing a DFS starting with `start`.
    */
-  def leafAt(leafIndex: Int, start: Node[K, V, T, A]): Option[LeafNode[K, V, T, A]] = {
+  def leafAt(leafIndex: Int, start: Node[K, V, T, A]): Option[LeafNode[K, V, T, A]] =
```

Comment:
I think currently the traversals we have are somewhat specialized to the tree structure itself, and can't easily accommodate searching for nodes by the leaf ID. For now I'd just as soon leave this in, although I guess it's also possible to make traversals a bit more general.